student/src/main/java/tds/student/web/handlers/TestResponseHandler.java

```diff
@@ -128,6 +137,17 @@ private void updateResponses (HttpServletRequest request, HttpServletResponse re
       // get the request information from tehe browser
       TestResponseReader responseReader = TestResponseReaderSax.parseSax (request.getInputStream (), testOpp);
 
+      List<Map<String,String>> responses = new ArrayList<>();
+      for(ItemResponseUpdate responseUpdate: responseReader.getResponses()) {
```

Comment:
That comment would be for any time you're populating data structures with bits of data from other objects.